dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -405,72 +408,70 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
-                            } catch (final IOException e1) {
-                                LOG.error("Failed to cancel job", e1);
-                            }
+            LOG.info("Drop event contains files");
+            final Ds3TreeTableValue value = selectedItem.getValue();
+            final String bucket = value.getBucketName();
+            final String targetDir = value.getDirectoryName();
+            LOG.info("Passing new Ds3PutJob to jobWorkers thread pool to be scheduled");
+            final String priority = (!savedJobPrioritiesStore.getJobSettings().getPutJobPriority().equals(resourceBundle.getString("defaultPolicyText"))) ? savedJobPrioritiesStore.getJobSettings().getPutJobPriority() : null;
+            final ImmutableList.Builder<Pair<String, Path>> builder = new ImmutableList.Builder<>();
+            ((List<Pair<String, String>>) db.getContent(DataFormat.lookupMimeType("local"))).stream()
+                    .map(pairStrings -> new Pair<>(pairStrings.getKey(), Paths.get(pairStrings.getValue())))
+                    .forEach(file -> {
+                        if (file.getValue().toFile().isDirectory() && file.getValue().toFile().listFiles().length == 0) {
+
+                            final CreateFolderTask task = new CreateFolderTask(session.getClient(), bucket, file.getKey(), loggingService, resourceBundle);
+                            workers.execute(task);
+                            task.setOnSucceeded(e -> {
+                                RefreshCompleteViewWorker.refreshCompleteTreeTableView(ds3Common, workers, loggingService);
+                                loggingService.logMessage("Created folder " + file.getKey(), LogType.INFO);
+                            });
+                            task.setOnFailed(e -> {
+                                loggingService.logMessage("Could not create folder " + file.getKey(), LogType.ERROR);
+                            });
+                        } else {
+                            builder.add(file);
                         }
-                        Ds3PanelService.refresh(selectedItem);
-                        ds3TreeTable.getSelectionModel().clearSelection();
-                        ds3TreeTable.getSelectionModel().select(selectedItem);
                     });
-                }
-            } else {
-                alert.showAlert(resourceBundle.getString("operationNotAllowedHere"));
+            final ImmutableList<Pair<String, Path>> pairs = builder.build();
+            if (pairs.isEmpty()) {
+                Ds3PanelService.refresh(selectedItem);
+                return;
             }
-            event.consume();
+            final Ds3PutJob putJob = new Ds3PutJob(session.getClient(), pairs, bucket, targetDir, jobInterruptionStore, priority, settingsStore.getProcessSettings().getMaximumNumberOfParallelThreads(), resourceBundle, settingsStore, loggingService, deepStorageBrowserPresenter, selectedItem);
```

Comment:
We should consider using a Guice Factory with AssistedInject to clean this up.  We should be able to reduce the number of dependencies that we have to pass into this presenter by moving this to a Guice Factory.